### ISSUE TITLE
[BUG] [C++] [Restbed] Restbed C++ OpenAPI Generator does not convert a vector of custom objects into a json string in the generated code

### ISSUE THREAD

**Issue by cornhedgehog** [OpenAPITools/openapi-generator#11605]
##### Description

Hi, I am using OpenAPI generator to generate Restbed C++ API server. Here is what I've noticed in the generated GET handler: for some reason resultObject never gets parsed, and the result is empty. Here is this handler from examples in [the OpenAPI Generator repository](https://github.com/OpenAPITools/openapi-generator/blob/master/samples/server/petstore/cpp-restbed/api/PetApi.cpp#L483):

```
void PetApiPetFindByStatusResource::handler_GET_internal(const std::shared_ptr<restbed::Session> session)
{
const auto request = session->get_request();

// Getting the query params

int status_code = 500;
std::vector<std::shared_ptr<Pet>> resultObject = std::vector<std::shared_ptr<Pet>>();
std::string result = "";

try {
    std::tie(status_code, resultObject) =
         handler_GET(status);
}
catch(const PetApiException& e) {
    std::tie(status_code, result) = handlePetApiException(e);
}
catch(const std::exception& e) {
    std::tie(status_code, result) = handleStdException(e);
}
catch(...) {
    std::tie(status_code, result) = handleUnspecifiedException();
}

if (status_code == 200) {

    const constexpr auto contentType = "application/json";
    //I want to return the result here but it would never happen since we are not using resultObject anywhere 
    returnResponse(session, 200, result.empty() ? "successful operation" : result, contentType);
    return;
}
if (status_code == 400) {

    const constexpr auto contentType = "text/plain";
    returnResponse(session, 400, result.empty() ? "Invalid status value" : result, contentType);
    return;
}
defaultSessionClose(session, status_code, result);
}
```

I can observe the same results for my project while using Restbed generator. I think the only way to fix it is to add some sort of result = resultObject.to_json_string() every time API is re-generated but this is not how it should work. Is there any workaround?

##### openapi-generator version
I am using version 5.4.0 which is the stable latest version.

##### OpenAPI declaration file content or url

Here is my .json spec file from my project that seems to be correct:

```
{
  "openapi": "3.0.0",
  "info": {
    "version": "1.0.0",
    "title": "My API"
  },
  "paths": {
    "/myobject": {
      "get": {
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "$ref": "#/components/schemas/MyObject"
                  }
                }
              }
            }
          },
          "default": {
            "description": "unexpected error"
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "MyObject": {
        "type": "object",
        "properties": {
          "prop1": {
            "type": "string"
          },
          "prop2": {
            "type": "number"
          }
        }
      }
    }
  }
}
```

##### Generation Details, Steps to Reproduce
I am using npx CLI tool to generate API using this command
`npx @openapitools/openapi-generator-cli generate -i apispec.json -g cpp-restbed-server -o ../OpenAPI/generatedapi`
The issue can be reproduced with help of Petstore example using Restbed generator, both for .json or .yaml spec.
`npx @openapitools/openapi-generator-cli generate -i petstore.yaml -g cpp-restbed-server -o /tmp/restbed-test`

In other words it seems to be easily reproduced. Please let me know if I can provide you with any more details.

**Comment by goto40** [comment]
I can confirm this behavior.
It seems that some code is generated which tries to call `std::vector<... >::toJsonString`, which does not exist.

Probably a special case, like for maps (`convertMapResponse(resultObject)`), needs to be inserted in case of arrays...

I can try to see if I manage to create a fix...

**Comment by goto40** [comment]
I played around and generated a fix to allow the pets example to work for the "/pets"-query:
`0001-quickfix-array2json.patch`:
```
From 69c83b43c7412c51614d116f623f390c087b688f Mon Sep 17 00:00:00 2001
From: goto40 <pierre.bayerl@googlemail.com>
Date: Tue, 21 Jun 2022 22:32:42 +0200
Subject: [PATCH] quickfix: array2json

---
 .../cpp-restbed-server/api-source.mustache    | 25 +++++++++++++++++--
 1 file changed, 23 insertions(+), 2 deletions(-)

diff --git a/modules/openapi-generator/src/main/resources/cpp-restbed-server/api-source.mustache b/modules/openapi-generator/src/main/resources/cpp-restbed-server/api-source.mustache
index 6a9c99ab284..eeb218378ef 100644
--- a/modules/openapi-generator/src/main/resources/cpp-restbed-server/api-source.mustache
+++ b/modules/openapi-generator/src/main/resources/cpp-restbed-server/api-source.mustache
@@ -76,6 +76,27 @@ std::string convertMapResponse(const std::map<KEY_T, VAL_T>& map)
     return result;
 }
 
+template<class Ptr>
+std::string convertResponse(const Ptr& resultObject) {
+    return resultObject->toJsonString();
+}
+
+template<class Ptr>
+std::string convertResponse(const std::vector<Ptr>& resultObject) {
+    std::ostringstream sstream;
+    std::string separator="";
+    sstream << "[\n";
+    for(const auto &ptr: resultObject) {
+        boost::property_tree::ptree item = ptr->toPropertyTree();
+        sstream << separator;
+        write_json(sstream, item);
+        separator=",\n";
+    }
+    sstream << "\n]\n";
+    std::string result = sstream.str();
+    return result;}
+
+
 {{#operation}}
 {{classname}}{{vendorExtensions.x-codegen-resource-name}}Resource::{{classname}}{{vendorExtensions.x-codegen-resource-name}}Resource(const std::string& context /* = "{{contextPath}}" */)
 {
@@ -201,7 +222,7 @@ void {{classname}}{{vendorExtensions.x-codegen-resource-name}}Resource::handler_
         result = resultObject;
         {{/isString}}
         {{^isString}}
-        result = resultObject->toJsonString();
+        result = convertResponse(resultObject);
         {{/isString}}
         {{/isModel}}
         {{#isMap}}
@@ -311,7 +332,7 @@ void {{classname}}{{vendorExtensions.x-codegen-resource-name}}Resource::handler_
         result = resultObject;
         {{/isString}}
         {{^isString}}
-        result = resultObject->toJsonString();
+        result = convertResponse(resultObject);
         {{/isString}}
         {{/isModel}}
         {{#isMap}}
-- 
2.25.1
```

I used the following code (main.cpp) in addition to the generated code:
```
#include <iostream>
#include "PetsApi.h"

struct MyPetsResource : org::openapitools::server::api::PetsApiPetsResource {
    std::pair<int, std::vector<std::shared_ptr<org::openapitools::server::api::Pet>>> handler_GET(int32_t const & limit) override {
        std::vector<std::shared_ptr<org::openapitools::server::api::Pet>> result;
        const char* names[] = {"Tom","Jerry","Anatol"};
        const char* tags[] = {"tag0","tag1","tag2"};
        for (size_t k=0;k<3;k++) {
            auto pet = std::make_shared<org::openapitools::server::api::Pet>();
            pet->setId(k);
            pet->setName(names[k]);
            pet->setTag(tags[k]);
            result.push_back(pet);
        }
        return {0, result}; // why do I have to return 0 and not 200 (=OK)?
    }
};

struct MyPetsPetIdResource : org::openapitools::server::api::PetsApiPetsPetIdResource {
};

int main() {
    std::cout << "restbed based server\n";

    auto service = std::make_shared<restbed::Service>(); // https://stackoverflow.com/questions/42559624/restbed-example
    org::openapitools::server::api::PetsApi api{service};
    api.setPetsApiPetsResource( std::make_shared<MyPetsResource>() );
    api.setPetsApiPetsPetIdResource( std::make_shared<MyPetsPetIdResource>() );

    auto settings = std::make_shared< restbed::Settings >( );
    settings->set_port( 1984 );
    settings->set_default_header( "Connection", "close" );
    service->start( settings );
}
```
With this I get a successful answer in my browser:
![Screenshot from 2022-06-21 22-29-00](https://user-images.githubusercontent.com/16887130/174892891-56f1abb3-73a8-413f-937f-dbe9cda67c6e.png)
 
@cornhedgehog I was puzzled why I had to return a return-code of 0 and not 200... Have you played with the `cpp-pistache-server` generator?
I am about to decide on which generator to focus my work...

**Comment by cornhedgehog** [comment]
@goto40 

> why I had to return a return-code of 0 and not 200

Maybe 0 is intended to show the absence of errors during the execution, and 200 code (or any other code) derives from the result. Have you defined return codes in the OpenAPI specification? (if you're using one).

I have not investigated `cpp-pistache-server`, it seems to be not stable enough for my purposes.
